doc/glossary.rst

```diff
@@ -285,6 +285,16 @@ General Concepts
         Note that in this case, the precision can be platform dependent.
         The `numeric` dtype refers to accepting both `integer` and `floating`.
 
+        In regards to the efficiency and precision of the NumPy array, the data type
+        (`dtype`) can make a significant impact. If one chooses a data type with
+        a higher precision like `np.float64` or `np.int64` it will result in slow
+        operations, and increase memory usage, but it will result in accurate results.
+        However, if one opts for lower precision types like `np.float32` or `np.int32`
```

Comment:
I don't thin that we need to repeat. We can just explicitly make the comparison between 64-bits and 32-bits numbers in the first sentence.